### PR TITLE
Added feature toggle and new column for encryption

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-228

### Change description ###

- This PR only deals with below

1. Adding a new column in letter db.

2. Adding feature toggle for enabling/disabling encryption.

3. Update existing code ton include is_encryption enabled field

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/StaleLettersTaskTest.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ public void should_not_pick_up_letter_if_it_is_already_printed() {
     }
 
     private Letter createLetterWithSentToPrintTime(LocalTime sentToPrintAtTime) {
-        Letter letter = new Letter(UUID.randomUUID(), randomMessageId(), "service", null, "type", null);
+        Letter letter = new Letter(UUID.randomUUID(), randomMessageId(), "service", null, "type", null,false);
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/StaleLettersTaskTest.java]: missing space

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 2]
```diff
@@ -93,14 +99,19 @@ private UUID saveNewLetter(ILetterRequest letter, String messageId, String servi
         );
 
         // TODO: encrypt zip content
+        boolean isEncrypted = false;
 
+        if (isEncryptionEnabled) {
+            isEncrypted = true;
+        }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: just use `isEncryptionEnabled`

File: src/test/java/uk/gov/hmcts/reform/sendletter/logging/AppInsightsTest.java [hunk 3]
```diff
@@ -120,7 +120,7 @@ public void should_track_ftp_delete() {
 
     @Test
     public void should_track_event_of_not_printed_letter() {
-        Letter letter = new Letter(UUID.randomUUID(), MESSAGE_ID, SERVICE_NAME, null, TYPE, null);
+        Letter letter = new Letter(UUID.randomUUID(), MESSAGE_ID, SERVICE_NAME, null, TYPE, null,false);
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/sendletter/logging/AppInsightsTest.java]: missing space

File: src/test/java/uk/gov/hmcts/reform/sendletter/services/zip/ZipFileNameHelperTest.java [hunk 4]
```diff
@@ -32,7 +32,7 @@ public void should_generate_expected_file_name() {
     @Test
     public void should_remove_underscores_from_service_name() {
         // given
-        Letter letter = new Letter(randomUUID(), randomUUID().toString(), "cmc_claim_store", null, "type", null);
+        Letter letter = new Letter(randomUUID(), randomUUID().toString(), "cmc_claim_store", null, "type", null,false);
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/sendletter/services/zip/ZipFileNameHelperTest.java]: and here

File: src/test/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTaskTest.java [hunk 5]
```diff
@@ -67,7 +67,7 @@ public void should_not_start_process_if_ftp_is_not_available() {
     }
 
     private Letter letterOfType(String type) {
-        return new Letter(UUID.randomUUID(), "msgId", "cmc", null, type, "hello".getBytes());
+        return new Letter(UUID.randomUUID(), "msgId", "cmc", null, type, "hello".getBytes(),true);
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTaskTest.java]: and here

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -64,3 +64,6 @@ tasks:
   mark-letters-posted: ${FTP_REPORTS_CRON:0 30 0 * * *}
   stale-letters-report: ${STALE_LETTERS_REPORT_CRON:0 30 1 * * *}
 
+encryption:
+  enabled: ${ENCRYPTION_ENABLED:false}
```
Comment (rkondratowicz) [inline src/main/resources/application.yaml]: update terraform

Comment (nitinprabhu) [reply src/main/resources/application.yaml]: Yes will be doing that as part of next PR when we integrate with service

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 7]
```diff
@@ -93,14 +99,14 @@ private UUID saveNewLetter(ILetterRequest letter, String messageId, String servi
...
-
         Letter dbLetter = new Letter(
             id,
             messageId,
             serviceName,
             mapper.valueToTree(letter.getAdditionalData()),
             letter.getType(),
-            zipContent
+            zipContent,
+            isEncryptionEnabled
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: this should always be `false` (for now). feature toggle doesn't do anything

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: If we use value from feature toggle we could then enable/disable it on specific env say sandbox and test it.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: but file is not actually being encrypted

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: Feel free to log something instead (when file is supposed to be encrypted)

**Pull request merged**